File: uri/go/1010.go
package main

import "fmt"

func main() {
	var p1,p2,q1,q2 int
	var v1,v2 float64
	fmt.Scan(&p1, &q1, &v1)
	fmt.Scan(&p2, &q2, &v2)
	fmt.Printf("VALOR A PAGAR: R$ %.2f\n", float64(q1)*v1+float64(q2)*v2)
}

File: uri/rust/1010.rs
fn get_float_vec() -> Vec<f64> {
    let mut buffer = String::new();
    std::io::stdin().read_line(&mut buffer).expect("Failed");
    let split = buffer.split_whitespace();
    let vec: Vec<&str> = split.collect();
    vec.iter().map(|x| x.trim().parse::<f64>().unwrap()).collect()
}

fn main() {
    let vec1: Vec<f64> = get_float_vec();
    let vec2: Vec<f64> = get_float_vec();

    println!("VALOR A PAGAR: R$ {:.2}", &vec1[1] * &vec1[2] + &vec2[1] * &vec2[2]);
}
